### PR TITLE
Feature - 3158 - Admin View Pool Page

### LINKED ISSUE #3158: Placeholder page for: View Pool
<sup>ℹ️ [Figma (root)][figma] | [Sitemap][sitemap]</sup>
<sup>_This initial comment is collaborative and open to modification by all._</sup>

## Description

This page contains static information from this pool, as well as buttons for other actions.
Only "Edit Pool Advertisement" button has a URL and Design

## Screenshot: Prototype
image

## Acceptance Criteria
- [ ] URL "https://talent.canada.ca/en/admin/pools/[ID]"
- [ ] Create whole form but not query for data
- [ ] Put behind feature flag "Direct Intake"
- [ ] 

[Wireframe ](https://www.figma.com/proto/b6lDM86QCJQ6iaAh7qrTYH/Admin-v4-(Direct-Intake)?page-id=0%3A1&node-id=1919%3A3386&viewport=-3442%2C1747%2C0.59&scaling=min-zoom&starting-point-node-id=1618%3A6064)

### PR DESCRIPTION
**Pull request by esizer**
Resolves #3158 

## Summary

This adds the new pool advertisement page to admin behind the direct intake feature flag.

## Testing

1. Ensure you enable the `FEATURE_DIRECTINTAKE` flag in `.apache_env`
2. Rebuild the php container
  1. `docker-compose stop php`
  2. `docker-compose rm php`
  3. docker-compose up --detach --build`
3. Navigate to `/admin/pools`
4. Click on the name of the pool
5. Ensure the page matches the [prototype](https://www.figma.com/proto/b6lDM86QCJQ6iaAh7qrTYH/Admin-v4-(Direct-Intake)?page-id=0%3A1&node-id=1919%3A3386&viewport=-3442%2C1747%2C0.59&scaling=min-zoom&starting-point-node-id=1618%3A6064)

## Note

The URL field just displays the title until we have on available.

## Screenshot
localhost_8000_en_admin_pools_081a27ce-8dab-4e4c-814b-003757cd60e9

### CODE REVIEW

**Comment by vd1992** [comment]
for anyone else who may look at this, and find that for some reason things won't load despite doing the php inject
try this handy tip courtesy of @petertgiles 

if that red line bit is less than 3 items, open the highlighted config.sjs in a separate tab and mash refresh a couple of times and things (hopefully) resolve
(this happened to me just now and the above fixed it)

image

**Review by vd1992 — COMMENTED** [review]
image

It is a bit odd to not see the entire pool advertisement URL thing, the arrowed bit, can it be lengthened? 
as well, marked with the vertical line, the three elements do not have equal height, per wireframe they look as they do 

wireframe ref
image

breadcrumbs also appears different in the wireframe
image

**Comment by Jerryescandon** [comment]
> 3\. leave as is

Leave as is.

The components in the design system should address this on a global scale, so we either fix for all or not. I think as a practice we should avoid doing any finagling with component design unless it is crucial to the interaction for some reason.

**Comment by Jerryescandon** [comment]
> We can do the full word but we _should_ avoid using all caps unless they are acronyms.

Fair enough. Sentence case it is. Will update the wireframe.

**Comment by esizer** [comment]
> It is a bit odd to not see the entire pool advertisement URL thing, the arrowed bit, can it be lengthened?

While the URL will likely be too long to be reasonably shown on most screen sizes, i have increased the width to more closely match the prototype with [increase url field width](https://github.com/GCTC-NTGC/gc-digital-talent/pull/3274/commits/df33e0c687bc32a7baf7cc8a61f11a66ead1bf71)

> breadcrumbs also appears different in the wireframe

Updated breadcrumbs to use ID rather than title in [updated breadcrumbs to use pool id](https://github.com/GCTC-NTGC/gc-digital-talent/pull/3274/commits/41a036061a170b3c0ce5b117c200d48866a3faa4)


> Fair enough. Sentence case it is. Will update the wireframe.

Thanks @Jerryescandon this has been fixed in [update short langs to full](https://github.com/GCTC-NTGC/gc-digital-talent/pull/3274/commits/a69d73a18f9015e3d040ce47862dc31ad779f075)

**Comment by petertgiles** [comment]
Regarding pool status, I assume since this is a pool advertisement page that you want the new derived pool advertisement status (DRAFT/PUBLISHED/CLOSED) rather than the older pool status (TAKING APPLICATIONS/NOT TAKING APPLICATIONS).  I thinking we should probably remove the old pool status field from the the pool application model in the API.

**Comment by petertgiles** [comment]
@Jerryescandon what is process number?  We have a field like that on pool candidate but not pool.
image

**Review by petertgiles — CHANGES_REQUESTED** [review]
Very nice!  I remembered to run Axe tools on it and 0 issues!  A few small things below but this looks 99% done.

File: frontend/admin/src/js/components/pool/ViewPool.tsx [hunk 1]
```diff
@@ -56,84 +154,468 @@ export const ViewPoolPage: React.FC<ViewPoolPageProps> = ({ pool }) => {
...
+                getAdvertisementStatus(pool.advertisementStatus ?? ""),
+              )}
+              label={intl.formatMessage({
+                defaultMessage: "Status",
+                description: "Label for a pool advertisements status",
+              })}
+            />
+          </Spacer>
+        </div>
+        <div data-h2-display="b(flex)" data-h2-align-items="b(flex-end)">
```
[the remark this answers is not in the document]
Comment (esizer) [reply frontend/admin/src/js/components/pool/ViewPool.tsx]: Good eye! 👀 

Fixed with [fix flex alignment](https://github.com/GCTC-NTGC/gc-digital-talent/pull/3274/commits/93947a4eabf39edb8a20aa13eb4857543cdfc9a6)

File: frontend/admin/src/js/components/pool/ViewPool.tsx [hunk 2]
```diff
@@ -1,42 +1,140 @@
...
+
+type SpacerProps = React.HTMLProps<HTMLSpanElement>;
+
+const Spacer = ({ children, ...rest }: SpacerProps) => (
+  <span data-h2-margin="b(bottom-right, s)" {...rest}>
+    {children}
+  </span>
+);
+
+const getLocalizedName = (
```
[the remark this answers is not in the document]
Comment (esizer) [reply frontend/admin/src/js/components/pool/ViewPool.tsx]: I was contemplating this, makes sense to me. Completed in [move localize name to common/helpers](https://github.com/GCTC-NTGC/gc-digital-talent/pull/3274/commits/d740ade28ebbdff9373d0a86b411e7da68f27d94)

File: frontend/admin/src/js/components/pool/ViewPool.tsx [hunk 3]
```diff
@@ -56,84 +154,468 @@ export const ViewPoolPage: React.FC<ViewPoolPageProps> = ({ pool }) => {
...
-        {pool.name && (
-          <h2
-            data-h2-margin="b(top-bottom, s) m(top-bottom, none)"
-            data-h2-font-weight="b(800)"
+        <Spacer>
+          <IconLink
+            mode="solid"
+            color="secondary"
+            type="button"
+            href="#"
```
[the remark this answers is not in the document]
Comment (petertgiles) [reply frontend/admin/src/js/components/pool/ViewPool.tsx:159]: Looks great!

File: frontend/admin/src/js/components/pool/ViewPool.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (petertgiles) [reply frontend/admin/src/js/components/pool/ViewPool.tsx]: Nice.

File: frontend/admin/src/js/components/pool/ViewPool.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (petertgiles) [reply frontend/admin/src/js/components/pool/ViewPool.tsx]: Very nice!

**Comment by Jerryescandon** [comment]
> what is process number? We have a field like that on pool candidate but not pool.

For this one I believe we have something that is call DCMO ID or something like that on profiles, which basically groups all the current applicants into "pools" however this number is currently being inputted into the OLD profile as a field. 

The proposal was to have an auto-generated key (process number) that replaces the DCMO ID that we can use to tie applicants to pools. I was expecting this now to be generated when the pool is created as opposed to letting DCMO input it.

thoughts?

File: frontend/admin/src/js/components/pool/ViewPool.tsx [hunk 4]
```diff
@@ -56,84 +154,468 @@ export const ViewPoolPage: React.FC<ViewPoolPageProps> = ({ pool }) => {
...
+        <div data-h2-display="b(flex)" data-h2-align-items="b(flex-end)">
+          <Spacer>
+            <Input
+              id="expiryDate"
+              name="expiryDate"
+              type="text"
+              readOnly
+              hideOptional
+              value={new Date(pool.expiryDate).toLocaleString(locale, {
+                dateStyle: "medium",
```
[the remark this answers is not in the document]
Comment (Jerryescandon) [reply frontend/admin/src/js/components/pool/ViewPool.tsx:352]: I think localized is fine

**Comment by petertgiles** [comment]
> The proposal was to have an auto-generated key (process number) that replaces the DCMO ID

Each pool has an auto-generated ID (like 3e3fff38-84d8-4c9d-a21e-1bec3ca84392) already.  The database uses this internally to associate an applicant to a pool.  Does this work for the use case?

We missed this in the database work that has been done already.  Can we kick this to a later sprint when we have a clear idea of the purpose of the field?

**Review by petertgiles — APPROVED** [review]

**Pull request merged**